### PR TITLE
add Sort transform

### PR DESCRIPTION
**Pull request by azylman**

### CODE REVIEW

File: transforms/sort.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+func (r *rows) Less(i, j int) bool {
+	return r.less(r.rows[i], r.rows[j])
+}
+func (r *rows) Swap(i, j int) {
+	r.rows[i], r.rows[j] = r.rows[j], r.rows[i]
+}
+
+// Sort takes in a function that reports whether the row i should sort before row j.
+// It outputs the rows in a stable, unsorted order.
```
Comment (d-sparks) [inline transforms/sort.go]: s/unsorted/sorted

Comment (azylman) [inline transforms/sort.go]: That's megaderp because it's not even stable

File: transforms/sort.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	return less
+}
+func (r *rows) Swap(i, j int) {
+	r.rows[i], r.rows[j] = r.rows[j], r.rows[i]
+}
+
+// Sort takes in a function that reports whether the row i should sort before row j.
+// It outputs the rows in sorted order. The sort is not guaranteed to be stable.
+func Sort(less func(i, j optimus.Row) (bool, error)) optimus.TransformFunc {
```
Comment (d-sparks) [inline transforms/sort.go]: What do you think about offering stable sort as well?  You could just define a Stable transformer and then either add a flag to this function or have two functions (transforms.Sort and transforms.Stable) that both call out to the logic within the current Sort function.

Comment (d-sparks) [inline transforms/sort.go]: (I only suggest it because it seems like an easy win.)

**Comment by d-sparks** [comment]
Beautiful code.  Can't find anything to complain about!

File: transforms/sort.go (hunk 2 above)
Comment (azylman) [inline transforms/sort.go]: I think it'd be better to do it as a separate function, and then I figure we can add it when we need it :)

**Pull request merged**